### PR TITLE
Add unlink slack account option to profile

### LINKED ISSUE #342: Allow user to unlink slack account
There is no way to change or unlink a slack account in the user's profile.

### PR DESCRIPTION
**Pull request by matiasb**
Fixes #342

### CODE REVIEW

File: engine/apps/api/views/user.py [hunk 1]
```diff
@@ -350,6 +352,13 @@ def get_telegram_verification_code(self, request, pk):
 
         return Response({"telegram_code": str(new_code.uuid), "bot_link": bot_link}, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=["post"])
+    def unlink_slack(self, request, pk):
+        user = self.get_object()
+        user.slack_user_identity = None
```
Comment (vadimkerr) [inline engine/apps/api/views/user.py:358]: Should we also delete the `SlackUserIdentity` instance in case there are no users associated with it?

**Review by vadimkerr — APPROVED** [review]
LGTM

File: engine/apps/api/views/user.py (hunk 1 above)
Comment (matiasb) [reply engine/apps/api/views/user.py:358]: I based this on the existing reset slack process (`ResetSlackView`), but also I think we don't want to lose track of any potential data (messages, etc) associated to the slack user?

Comment (vadimkerr) [reply engine/apps/api/views/user.py:358]: Makes sense 👍

**Pull request merged**